### PR TITLE
Refactor Gangway code into an internal pkg allowing for easier testing

### PR DESCRIPTION
**Pull request by stevesloka**
Signed-off-by: Steve Sloka <steves@heptio.com>

### CODE REVIEW

File: internal/session/session.go [hunk 1]
```diff
@@ -24,26 +24,34 @@ import (
...
-func initSessionStore() {
-
-	sessionStore = sessions.NewCookieStore(generateSessionKeys())
-}
-
-func cleanupSession(w http.ResponseWriter, r *http.Request) {
-
-	session, err := sessionStore.Get(r, "gangway")
+// CleanupSession removes the current session from the store
+func (s *Session) CleanupSession(w http.ResponseWriter, r *http.Request) {
```
Comment (alexbrand) [inline internal/session/session.go]: nit: Stutter
```suggestion
func (s *Session) Cleanup(w http.ResponseWriter, r *http.Request) {
```

Comment (alexbrand) [reply internal/session/session.go]: I have also seen `Destroy` more frequently used in this context.

File: cmd/gangway/main.go [hunk 2]
```diff
@@ -16,27 +16,27 @@ package main
...
+	"github.com/heptiolabs/gangway/internal/config"
+	"github.com/heptiolabs/gangway/internal/oidc"
+	"github.com/heptiolabs/gangway/internal/session"
 	"github.com/justinas/alice"
 	log "github.com/sirupsen/logrus"
 	"golang.org/x/oauth2"
 )
 
-var cfg *Config
+var cfg *config.Config
```
[the remark this answers is not in the document]
Comment (stevesloka) [reply cmd/gangway/main.go]: Yup I agree, was just taking baby steps and could address in a future pr.

**Comment by stevesloka** [comment]
@alexbrand I addressed all comments, thanks!

**Pull request merged**